### PR TITLE
Restricted device UI

### PR DESCRIPTION
**Pull request by Mark-Powers**
This adds a gray background to device rows in the calendar that have the "authorized_projects" capability when viewing from a project that does not have authorization to use the device. When hovering over the row, a box appears describing that this device is restricted.

image

### CODE REVIEW

**Comment by Mark-Powers** [comment]
If a device has the extra capability `restricted_reason`, this is now shown in the tooltip. 

image

File: blazar_dashboard/content/leases/views.py [hunk 1]
```diff
@@ -68,6 +68,16 @@ def calendar_data_view(request, resource_type):
     resources, reservations = api_mapping[resource_type](request)
     data['resources'] = resources
     data['reservations'] = reservations
+    # Include authorized_projects capability for devices in response
+    if resource_type == "device":
+        data["project_id"] = request.user.project_id
+        for device in resources:
+            full_device = api.client.device_with_capabilities(request, device["id"])
```
Comment (diurnalist) [inline blazar_dashboard/content/leases/views.py]: This turns this into an O(N) operation and we pay overhead on these requests. Do the capabilities not come back on the device list request? Is it possible to include them on the device list request via some flag, if they are not returned by default?

**Review by diurnalist — CHANGES_REQUESTED** [review]
See if we can collapse this into O(1), if the list of devices grows the calendar will be harder to show over time.

**Comment by Mark-Powers** [comment]
I see why I initially thought extra capabilities were not shown in the device list, but I see they were filtered out. This is updated.

**Review by diurnalist — APPROVED** [review]
Excellent!

**Pull request merged**